.github/workflows/reusable-build.yml

```diff
@@ -55,12 +55,19 @@ jobs:
       LD_FLAGS: ${{inputs.ld_flags}}
 
     steps:
+    - id: skip_check
```

Comment:
Can we make this as its own workflow that CI/CD calls first to test if build and test is needed? Having to modify every step in the child workflows seems cumbersome. 

1) Add a "check if build needed workflow"
2) Add a dependency from build -> "check if build needed" 
3) Add condition to test output from "check if build needed"

This would then only require modifying CI/CD and wouldn't require updating each step in the child workflows.

Reply:
As per offline discussion this causes required checks to not run, which blocks PR's from merging.

Reply:
That won't work because the branch protection has a list of required jobs that must succeed to allow a PR to be merged.  So if they're not run then nothing can be merged.